code/modules/fish/fish_eggs.dm

```diff
@@ -5,81 +5,70 @@
 	icon = 'icons/obj/fish_items.dmi'
 	icon_state = "eggs"
 	w_class = 2
-	var/fish_type = null			//Holds the name of the fish that the egg is for
+	var/datum/fish/fish_type = null			//Holds the datum of the fish that the egg is for, null means dud eggs
 
 /obj/item/fish_eggs/New()
 	..()
-
-var/global/list/fish_eggs_list = list("dud" = /obj/item/fish_eggs,
-									"goldfish" = /obj/item/fish_eggs/goldfish,
-									"clownfish" = /obj/item/fish_eggs/clownfish,
-									"shark" = /obj/item/fish_eggs/shark,
-									"baby space carp" = /obj/item/fish_eggs/babycarp,
-									"catfish" = /obj/item/fish_eggs/catfish,
-									"feederfish" = /obj/item/fish_eggs/feederfish,
-									"salmon" = /obj/item/fish_eggs/salmon,
-									"shrimp" = /obj/item/fish_eggs/shrimp,
-									"electric eel" = /obj/item/fish_eggs/electric_eel,
-									"glofish" = /obj/item/fish_eggs/glofish,
-									)
+	if(fish_type)
+		fish_type = new fish_type
```

Comment:
Going to have to `qdel` the `fish_type` in a `Destroy` if you're going to have this act as a holder.